mcs/class/System/System.Diagnostics/PerformanceCounterCategory.cs

```diff
@@ -147,14 +168,24 @@ public static bool CounterExists (string counterName, string categoryName)
 
 		// may throw ArgumentNullException, InvalidOperationException
 		// (categoryName is "", machine name is bad), Win32Exception
-		public static bool CounterExists (string counterName, string categoryName, string machineName)
+		unsafe public static bool CounterExists (string counterName, string categoryName, string machineName)
 		{
 			if (counterName == null)
 				throw new ArgumentNullException ("counterName");
 			CheckCategory (categoryName);
 			if (machineName == null)
 				throw new ArgumentNullException ("machineName");
-			return CounterCategoryExists (counterName, categoryName, machineName);
+
+			if (!ValidMachine (machineName))
+				return false;
+
+			unsafe {
+				fixed (char* fixed_counterName = (counterName != null) ? counterName : null,
```

Comment:
The `fixed_counterName = (counterName != null) ? counterName : null` is done automatically with `fixed_coutnerName = counterName` so no need to use this ternary operator.

Reply:
Ok, I didn't realize that.